### PR TITLE
Netsuite: Support additional files

### PR DESCRIPTION
**Pull request by omrilit**
Some context:
In SDF, emailtemplate & advancedpdftemplate types may have another file in the project containing the template's content. In the xml there is no mention for that content and the way to understand the connection between the content file and the xml is by the files names (e.g. `email_template_script_id.xml` & `email_template_script_id.template.html`).

In this PR:
- detect the additional files suffix
- add a 'content' field to types with additional file
- fetch the content from the additional file to the `content` field and set it as a `StaticFile`
- deploy the `content` value to a separate file in the project upon add & update
- support resolve & restore of static files in the adapter

### CODE REVIEW

File: packages/netsuite-adapter/src/types/custom_types/emailtemplate.ts [hunk 1]
```diff
@@ -93,6 +93,11 @@ export const emailtemplate = new ObjectType({
       annotations: {
       },
     }, /* Original description: The default value is F. */
+    content: {
```
Comment (adamro) [inline packages/netsuite-adapter/src/types/custom_types/emailtemplate.ts:96]: Instead of having a "Object level" annotation `ADDITIONAL_FILE_SUFFIX` that in-directly relates to this field, you can have this annotation on this field and make the type `ADDITIONAL_FILE` (which will also help you transform the content value to AdditionalFile)

File: packages/netsuite-adapter/src/client/client.ts [hunk 2]
```diff
@@ -210,27 +221,32 @@ export default class NetsuiteClient {
...
-  private static async deploy(filePath: string, customizationInfo: CustomizationInfo,
-    executor: CommandActionExecutor): Promise<void> {
-    await writeFile(filePath, convertToXmlContent(customizationInfo))
-    await NetsuiteClient.executeProjectAction(COMMANDS.ADD_PROJECT_DEPENDENCIES, {}, executor)
-    await NetsuiteClient.executeProjectAction(COMMANDS.DEPLOY_PROJECT, {}, executor)
+    const dirPath = NetsuiteClient.getObjectsDirPath(project.projectName)
+    await writeFile(path.resolve(dirPath, `${filename}.xml`), convertToXmlContent(customizationInfo))
+    if (!_.isUndefined(customizationInfo.fileContent)) {
+      await writeFile(path.resolve(dirPath,
+        `${filename}${ADDITIONAL_FILE_PATTERN}${customizationInfo.fileContent.extension}`),
```
Comment (adamro) [inline packages/netsuite-adapter/src/client/client.ts:244]: `fileContent.extension` can be undefined according to line 63 and it doesn't seem to be handled well here
can it actually be undefined?

File: packages/netsuite-adapter/src/transformer.ts [hunk 3]
```diff
@@ -81,14 +81,26 @@ export const createInstanceElement = (values: Values, type: ObjectType):
   const transformAttributeKey: MapKeyFunc = ({ key }) =>
     (key.startsWith(ATTRIBUTE_PREFIX) ? key.slice(ATTRIBUTE_PREFIX.length) : key)
 
-  const valuesWithTransformedAttrs = mapKeysRecursive(values, transformAttributeKey)
+  const valuesWithTransformedAttrs = mapKeysRecursive(customizationInfo.values,
+    transformAttributeKey)
+  const instanceName = getInstanceName(valuesWithTransformedAttrs)
+  const fileContentField = Object.values(type.fields)
```
Comment (adamro) [inline packages/netsuite-adapter/src/transformer.ts:87]: you might want to write somewhere that you have an assumptions there's only one line this per object cause it the code is based on it in a few places

File: packages/netsuite-adapter/src/transformer.ts [hunk 4]
```diff
@@ -170,8 +182,24 @@ export const toCustomizationInfo = (instance: InstanceElement): CustomizationInf
...
+  const customizationInfo: CustomizationInfo = { typeName, values }
+
+  const fileContentField = Object.values(instance.type.fields)
+    .find(f => isPrimitiveType(f.type) && f.type.isEqual(fieldTypes.fileContent))
+  if (!_.isUndefined(fileContentField) && !_.isUndefined(values[fileContentField.name])) {
+    customizationInfo.fileContent = {
+      extension: fileContentField.annotations[ADDITIONAL_FILE_SUFFIX],
+      content: values[fileContentField.name],
+    }
+    delete values[fileContentField.name]
```
Comment (adamro) [inline packages/netsuite-adapter/src/transformer.ts:199]: why is this needed?

Comment (omrilit) [reply packages/netsuite-adapter/src/transformer.ts:199]: This value should be only in the fileContent and not in the CustomizationInfo.values

File: packages/netsuite-adapter/src/transformer.ts (hunk 3 above)
Comment (omrilit) [reply packages/netsuite-adapter/src/transformer.ts:87]: Will add UT to verify that there is max one fieldTypes.fileContent field in each customObject type
